### PR TITLE
Changes email activation logic

### PR DESCRIPTION
**Pull request by Moejoe90**

### CODE REVIEW

File: app/components/emailVerify.tsx [hunk 1]
```diff
@@ -1,40 +1,24 @@
...
 }
 
-interface IEmailVerifyStateProps {
-  isVerified?: boolean;
-}
-
-export const emailVerifyComponent: React.SFC<
-  IEmailVerifyDispatchProps & IEmailVerifyStateProps
-> = ({ verifyEmail, isVerified, goHome }) => {
-  if (isVerified) {
```
Comment (krzkaczor) [inline app/components/emailVerify.tsx]: this kind of logic should not be in react component anyway 👍

File: app/modules/auth/sagas.ts [hunk 2]
```diff
@@ -55,7 +58,14 @@ export async function verifyUserEmailPromise({
   getState,
   notificationCenter,
 }: TGlobalDependencies): Promise<void> {
-  const userCode = selectActivationCodeFromQueryString(getState().router);
+  const state = getState();
+  const userCode = selectActivationCodeFromQueryString(state.router);
+  const urlEmail = selectLightWalletEmailFromQueryString(state.router);
+  const verifiedEmail = selectVerifiedUserEmail(state.auth);
+  if (urlEmail === verifiedEmail) {
+    notificationCenter.error("Your email is already verified");
```
Comment (krzkaczor) [inline app/modules/auth/sagas.ts]: i think it should be just info notification

File: app/modules/auth/sagas.ts [hunk 3]
```diff
@@ -55,7 +58,14 @@ export async function verifyUserEmailPromise({
   getState,
   notificationCenter,
 }: TGlobalDependencies): Promise<void> {
-  const userCode = selectActivationCodeFromQueryString(getState().router);
+  const state = getState();
```
Comment (krzkaczor) [inline app/modules/auth/sagas.ts]: you should never use getState injected function. Get data first in saga using select effect and pass it to promise returing function as argument. In this case extract fi

File: app/components/emailVerify.tsx (hunk 1 above)
Comment (Moejoe90) [reply app/components/emailVerify.tsx]: True that's why i took it off

File: app/modules/auth/sagas.ts (hunk 2 above)
Comment (Moejoe90) [reply app/modules/auth/sagas.ts]: done

**Pull request merged**